models/amity.py

```diff
@@ -0,0 +1,655 @@
+                room_type = "living-space"
+            if room_name.lower() not in ([room.name.lower() for room in self.get_all_rooms()]):
+                try:
+                    if room_type in self.allowed_living_space_strings:
+                        new_living_space = LivingSpace(room_name)
+                        self.living_spaces.append(new_living_space)
+                        new_rooms.append(new_living_space)
+                    else:
+                        new_office = Office(room_name)
+                        self.offices.append(new_office)
+                        new_rooms.append(new_office)
+                except TypeError as e:
+                    raise(e)
+                except Exception as e:
+                    print(e)
+                    raise(e)
+            else:
+                return self.return_error_for_printing(self.error_codes[3] + " '%s'" % room_name)
+        return new_rooms
+
+    def add_person(self, first_name, last_name, type, wants_accommodation=False):
+        if not isinstance(wants_accommodation, bool):
+            return self.return_error_for_printing(self.error_codes[7] + " '%s'" % wants_accommodation)
+        try:
+            new_person = None
+            if type.lower() in self.allowed_fellow_strings:
+                new_person = Fellow(first_name, last_name)
+                new_person.wants_accommodation = wants_accommodation
+                self.fellows.append(new_person)
+            elif type.lower() in self.allowed_staff_strings:
+                new_person = Staff(first_name, last_name)
+                self.staff.append(new_person)
+            else:
+                return self.return_error_for_printing(self.error_codes[5] + " '%s'" % type)
+
+            # Randomly allocate office to new person
+            if not self.offices:
+                print("\nThere exists no offices to assign to '%s %s'" % (new_person.first_name, new_person.last_name))
+            else:
+                self.randomly_allocate_room(new_person, self.allowed_office_strings[0])
+
+            if wants_accommodation and type.lower() in self.allowed_fellow_strings:
+                # Randomly assign available living space to fellow
+                if not self.living_spaces:
+                    print("\nThere exists no living space to assign fellow to")
+                else:
+                    result = self.randomly_allocate_room(new_person, self.allowed_living_space_strings[0])
+                if result:
+                    # Reset wants accommodation to False since they now have accommodation
+                    new_person.wants_accommodation = False
+
+            return new_person
+
+        except TypeError as e:
+            raise (e)
+        except Exception as e:
+            print(e)
+            raise (e)
+
+    def allocate_room_to_person(self, person, room, reallocate=False):
+        try:
+            if room.get_max_occupants() - room.num_of_occupants:
```

Comment:
This would have been better as a method in your room class to check room availability e.g. `if room.is_available()`